test/pre-need/config/sponsorMailingAddress.unit.spec.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { mount } from 'enzyme';
+
+import { DefinitionTester, fillData } from '../../util/schemaform-utils.jsx';
+import formConfig from '../../../src/js/pre-need/config/form';
+
+describe('Pre-need sponsor mailing address', () => {
+  const { schema, uiSchema } = formConfig.chapters.contactInformation.pages.sponsorMailingAddress;
+  it('should render', () => {
+    const form = mount(
+      <DefinitionTester
+        schema={schema}
+        definitions={formConfig.defaultDefinitions}
+        uiSchema={uiSchema}/>
+    );
+
+    expect(form.find('input').length).to.equal(4);
+    expect(form.find('select').length).to.equal(2);
+  });
+
+  it('should submit empty form', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        schema={schema}
+        definitions={formConfig.defaultDefinitions}
+        onSubmit={onSubmit}
+        uiSchema={uiSchema}/>
+    );
+
+    form.find('form').simulate('submit');
+
+    expect(form.find('.usa-input-error').length).to.equal(0);
+    expect(onSubmit.called).to.be.true;
+  });
+
+  it('should submit with valid data', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        schema={schema}
+        definitions={formConfig.defaultDefinitions}
+        onSubmit={onSubmit}
+        uiSchema={uiSchema}/>
+    );
+
+    fillData(form, 'input#root_application_veteran_address_street', 'Test');
```

Comment:
Should we consider creating reusable constants for these commonly used values (such as "Test"), or is it better to have the values exposed/visible in the tests themselves for maintainability?

Reply:
I don't think the constants buys us much, but we could potentially use the data in the json files for the schema tests.